aten/src/ATen/core/Tensor.cpp

```diff
@@ -92,9 +93,183 @@ const Tensor& Tensor::requires_grad_(bool _requires_grad) const {
+      return "alias";
+    case ViewMeta::Type::view_as_real:
+      return "view_as_real";
+    case ViewMeta::Type::view_as_complex:
+      return "view_as_complex";
+    case ViewMeta::Type::_conj:
+      return "_conj";
+    case ViewMeta::Type::_neg_view:
+      return "_neg_view";
+    case ViewMeta::Type::chunk:
+      return "chunk";
+    case ViewMeta::Type::diagonal:
+      return "diagonal";
+    case ViewMeta::Type::diagonal_Dimname:
+      return "diagonal_Dimname";
+    case ViewMeta::Type::expand_as:
+      return "expand_as";
+    case ViewMeta::Type::narrow:
+      return "narrow";
+    case ViewMeta::Type::narrow_Tensor:
+      return "narrow_Tensor";
+    case ViewMeta::Type::movedim_intlist:
+      return "movedim_intlist";
+    case ViewMeta::Type::movedim_int:
+      return "movedim_int";
+    case ViewMeta::Type::reshape_as:
+      return "reshape_as";
+    case ViewMeta::Type::detach:
+      return "detach";
+    case ViewMeta::Type::transpose_int:
+      return "transpose_int";
+    case ViewMeta::Type::transpose_Dimname:
+      return "transpose_Dimname";
+    case ViewMeta::Type::unsqueeze:
+      return "unsqueeze";
+    case ViewMeta::Type::_values:
+      return "_values";
+    case ViewMeta::Type::indices:
+      return "indices";
+    case ViewMeta::Type::view:
+      return "view";
+    case ViewMeta::Type::view_dtype:
+      return "view_dtype";
+      default:
+        return "UNKNOWN_VIEW_TYPE";
+  }
+}
+
+std::ostream& operator<<(std::ostream& str, ViewMeta::Type rhs) {
+  return str << ViewMetaTypeToString(rhs);
+}
+
+Alias::Alias(const at::Tensor& base) : base_(base.clone()) {
+  auto impl = dynamic_cast<at::FunctionalTensorImplBase*>(base_.unsafeGetTensorImpl());
+  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(impl != nullptr);
+  // Note [Marking Alias Tensors]
+  // In the functionalization pass, calling a view operation on a tensor forks it off an Alias object
+  // which is shared between the original tensor and the view.
+  // The Alias object contains a base tensor, which is used to sync operations across the view tensors.
+  // Throughout the Alias' life cylc,e we need to perform various views and reshapes on the base alias tensor.
+  // We don't want that process to recursively fork off another Alias object though, so we explicitly mark
+  // the base tensor as an alias.
+  // Another option would have been to implement separate {view}_copy() kernels for every corresponding view op,
+  // which have fallthrough kernels registered in the functionalization pass.
+  impl->mark_as_alias();
+}
+
+const at::Tensor& Alias::base() const {
+  return base_;
+}
+
+// metas is taken by value on purpose - we want to copy the vector.
+void Alias::add_update(const at::Tensor& updated_val, std::vector<ViewMeta> metas) {
+  updates_.push_back({updated_val, metas});
+  generation_++;
+}
+
+void Alias::apply_update(const Update& update) {
+  // TODO: Handle the other important view ops. This requires computing their inverses.
```

Comment:
For landing it seems like the XLA set would be good.

Reply:
We should also consider falling back to `as_strided` if there isn't explicit support for a view operation.